### PR TITLE
Add editorconfig for TS projects

### PR DESCRIPTION
**Pull request by SiAdcock**

### CODE REVIEW

File: ts-server/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+; top-most EditorConfig file
+root = true
+
+[*]
+indent_style = tab
```
Comment (SiAdcock) [inline ts-server/.editorconfig:5]: For [accessibility reasons](https://alexandersandberg.com/tabs-for-accessibility/)

File: ts-server/.editorconfig [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+indent_size = 4
```
Comment (SiAdcock) [inline ts-server/.editorconfig:6]: Happy to compromise on this!

File: ts-server/.editorconfig [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[package.json]
+indent_size = 2
```
Comment (SiAdcock) [inline ts-server/.editorconfig:13]: `npm` autoformats `package.json` every time you install a new dependency.

File: ts-server/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+[*.yml]
+indent_style = space
+indent_size = 2
```
Comment (SiAdcock) [inline ts-server/.editorconfig:17]: Because `yml`

File: ts-server/.editorconfig [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+[*.md]
+indent_style = space
```
Comment (SiAdcock) [inline ts-server/.editorconfig:20]: Using tabs to indent pre-formatted code in markdown makes the spacing too extreme when rendered on GitHub, so we revert back to spaces here.

**Comment by sndrs** [comment]
this is the default in the repo now, closing

**Pull request closed without merging**